import json
import multiprocessing

def create_managed_params(_manager: multiprocessing.Manager, _params: dict):
    managed_dict = _manager.dict()
    for param in _params.items():
        managed_dict[param[0]] = 0
    print(f"Loaded params: {json.dumps(_params)}")
    print(f"Initialized params: {managed_dict}")

    return managed_dict

lst_param = {
    #
    'MOTOR_LEFT_TEMPERATURE': '3b23507c-09c5-40c2-93c5-4be21d461a98',
    'MOTOR_RIGHT_TEMPERATURE': '7de01773-2efa-4e1a-8bbc-3c55a6fd6174',
    'CONTROLLER_LEFT_TEMPERATURE': '0d865620-ab05-4394-8e61-5b0e8585346d',
    'CONTROLLER_RIGHT_TEMPERATURE': 'c611cd24-c07f-46d0-8826-3e781e9460d5',
    'BMS_VOLTAGE_CELL_1': '8e5e26d3-a18e-4ebf-aba9-e8338767ba2d',
    'BMS_VOLTAGE_CELL_2': '541b5344-d88b-43d2-a404-c000c0e37f39',
    'BMS_VOLTAGE_CELL_3': '8a1ed91c-c4c7-4fe1-b6a8-7662d219807c',
    'BMS_VOLTAGE_CELL_4': 'd4423b1d-0b90-42f9-b807-6bc62e67b90c',
    'BMS_TEMPERATURE_CELL_1': 'd714db81-459c-4e4f-8dab-c7edf4dab57e',
    'BMS_TEMPERATURE_CELL_2': '26c5350f-abc2-4283-a974-26232f234b05',
    'BMS_TEMPERATURE_CELL_3': '1d0fe152-1b0b-42fd-9829-26819f461af4',
    'BMS_TEMPERATURE_CELL_4': 'a45079b3-befa-4055-b855-550fdc4c9cb6',
    'MPPT_1_Current_IN': '02cccbae-61fc-4ca0-a094-e9816f670d1b',
    'MPPT_1_Voltage_IN': '3eb1dbb0-e600-48dd-9a81-da1fa27a36c4',
    'MPPT_1_Voltage_OUT': 'd3e0f6b1-4c27-440a-b4fd-09985f6c63f2',
    'MPPT_1_Power_IN': 'e3df95b6-ecca-4dc8-b38a-0e1145e8ab11',
    'MPPT_1_PCB_TEMP': 'f53bc1b6-41ff-4e9a-b970-58045d87cb3a',
    'MPPT_1_MOSFET_TEMP': '94dde6de-61c0-4a2e-80ed-558b02354cc6',
    'MPPT_2_Current_IN': '2fc49174-558e-4399-9842-dbe5340e9018',
    'MPPT_2_Voltage_IN': '4a372a35-98ae-4a43-af59-a049862df815',
    'MPPT_2_Voltage_OUT': '77bdfc94-c991-4b12-b999-6ffdbadbfcb0',
    'MPPT_2_Power_IN': '086555db-3d30-4586-8878-ead1e6f4c662',
    'MPPT_2_PCB_TEMP': 'ca9d2371-f1b3-4ca6-a32e-02d7b51ee1fd',
    'MPPT_2_MOSFET_TEMP': '0975697d-52cd-4939-9b81-43a8ae2cf711',
    'MPPT_3_Current_IN': '7c073d42-179e-4b8d-84e0-6c6d35806fd9',
    'MPPT_3_Voltage_IN': 'a5acbc55-b6dd-491e-888d-1a90a59c5edb',
    'MPPT_3_Voltage_OUT': '481df9e3-ba22-4cde-9f6f-9a20372dc390',
    'MPPT_3_Power_IN': 'af7dfd60-bd89-4cf5-a3f6-a0e9c919c45c',
    'MPPT_3_PCB_TEMP': 'e3b96392-959d-4ab7-b677-e97bda9ced1c',
    'MPPT_3_MOSFET_TEMP': '288a1ec4-c95d-4dde-a0f3-7af088c4537b',
    'MPPT_4_Current_IN': '198efe34-5138-48c0-a1a9-0f4df0146a19',
    'MPPT_4_Voltage_IN': '53a8815c-36fa-45d1-8a24-c70652b76340',
    'MPPT_4_Voltage_OUT': 'e4b7934b-08a6-495e-907b-1da1436d9a41',
    'MPPT_4_Power_IN': 'f48f457e-dfa0-4db9-8e56-3828af25b2b9',
    'MPPT_4_PCB_TEMP': 'e3767d4c-6141-43a7-9cdd-c45a36c55da6',
    'MPPT_4_MOSFET_TEMP': '8c173557-eac4-4c0f-94c2-fad835ec3c2f',
    'BMS_AVG_CELL_VOLTAGE': 'ac53cac6-49d6-4d9a-b25d-8cc0d389b136',
    'BMS_CELL_W_LOWEST_VOLTAGE': '014980de-cc8a-4783-95f3-d7ab21470836',
    'BMS_CELL_VOLTAGE_MIN': '4a980bcd-6f29-4f42-88df-8044878f677a',
    'BMS_CELL_W_HIGHEST_VOLTAGE': 'ec2c96f7-c9a4-4bb1-a711-d6997e388497',
    'BMS_CELL_VOLTAGE_MAX': 'f4525ebe-c625-4918-a263-f6fa3af61ea2',
    'BMS_ERROR_FLAGS': '780a6e5c-b919-44c7-ae14-b88627fa5800',
    'BMS_WARNING_FLAGS': 'febd89d3-0283-4a01-85e0-13c76e102c1b',
    'BMS_PACK_VOLTAGE': '9d7b9e3a-ed6d-4e39-9784-3e1df46a4e51',
    'BMS_PACK_CURRENT': '7c22ef5c-ac3f-418a-9c5d-251b3e53be30',
    'BMS_SYSTEM_STATE': '90bac425-f0ab-4cde-bd1f-9319da7d3992',
    'BMS_SOC': 'b9d0d66d-be77-415e-99e7-accbd2e8191e',
    'BMS_CHARGER_ENABLED': 'a1053031-239f-49c5-97f9-f9600ac428c5',
    'BMS_PACK_TEMP_HIGH': 'fc9c5fbf-9583-42be-aeba-979b1df1cc86',
    'BMS_LMU_W_HIGHEST_TEMP': 'd5bc8ddc-b498-4695-aca8-132dd41d0d29',
    'BMS_PACK_TEMP_LOW': 'fd530c22-b727-45d6-b991-f465372de19e',
    'BMS_LMU_W_LOWEST_TEMP': 'b55c5f90-b1bb-4e7f-8c20-fe593d5a7c98',
    'BMS_REMAINING_CHARGE_TIME': '1d7d1659-ef88-4abe-98cb-0787bf2efd02',
    'BMS_REMAINING_ENERGY': 'd82bc8a4-6cec-41d3-815d-9dc56b3bc224',
    'BMS_SOH': 'a9cc4ee7-53ff-4be3-9804-0ebc6c05cb0b',
    'BMS_CELLS_CONNECTED': '6f1a1b8e-66e1-4bdb-a643-b641b2bdbb49',
    'BMS_POST_PROC_2': '1cef16d6-24a8-4d03-8d04-4b1ddb689555',
    'BMS_MAIN_REL_POS': 'ddc40657-64d3-4415-a702-05fb3674b92a',
    'BMS_MAIN_REL_NEG': '8bf05e64-1735-4570-8b3c-5bf10b4ae0b4',
    'BMS_CHARGE_EN': '3e1398da-fb7f-4f9c-aafa-68857bc76fba',
    'BMS_PRECHARGE_REL': 'f8f083c3-51c4-474d-9839-0055e5944b10',
    'BMS_DISCHARGE_EN': 'a050a5c1-734a-41a5-88ec-a869ad2053e7',
    'BMS_ERROR_LED': '704eed40-6e74-4215-9f29-e8658b903f19',
    'BMS_FAN_CTRL_1': '1d5ffc43-b1a1-4d19-8a95-6c39bc227fc6',
    'BMS_FAN_CTRL_2': 'd047e17e-2867-4d1a-b2e3-128e4240c6d5',
    'BMS_PACK_C_RATE': 'bc25a949-5940-4cfb-99ca-b11e08fbca02',
    'OVARTECH_MAX_DC_V': '58542f7b-8435-4064-9cba-5bbcfca1abae',
    'OVARTECH_MAX_DC_I': '3f338491-85a8-4b09-835e-69cda16028ce',
    'OVARTECH_CONTROL': 'ac8d57da-a27a-4d89-9365-3e4d379ef032',
    'STOP': '0e8bfdb0-e7a7-424e-a2a3-f9296e250630',
    'LOW_BEAM': '389dad70-04a9-4a94-878e-9fc3b11e700a',
    'HIGH_BEAM': '8d146375-d967-44b1-b49f-2b56f423be47',
    'RIGHT_INDICATOR': 'd3120098-1708-4df0-be34-1278bd55a710',
    'LEFT_INDICATOR': '63bbae56-8f61-43ec-86bc-c16cf2d0301f',
    'PARK': '50b641f5-07fa-4bf7-a546-eda2d35ed998',
    'HAZARD': '64ff6e7d-589c-426b-afed-81da556bf2f5',
    'HORN': '986ff37e-a3ad-4244-bc75-39f7a1ea4e9a',
    'HANDBRAKE': '3f676239-3253-43ac-bd31-ca294cc14336',
    'BATT_ERROR': '264155bf-1b35-4c2d-bfb6-50cfb780c828',
    'ENGINE_ERROR': '7dfd341e-9133-4917-8273-e7d4497a3c9f',
    'DRIVEMODE': 'a3c2927c-e47d-4f80-af84-ecded61d187e',
    'INTERIOR': 'c3926352-743b-4925-b541-03eeac0513c5',
    'CRUISE_ON': 'd0dbc0d8-a5bf-428e-8907-85de4b930b3c',
    'CRUISE_ACTIVE': 'cd0676fc-10bd-4157-8ad7-bdbac1adb5b3',
    'THROTTLE': '23f7a098-d33a-4764-83db-a33fea746d5f',
    'MAIN_SWITCH': '44164b22-c984-4abe-9486-62766a7c70f4',
    'DIRECTION': 'f02cf450-8081-4e61-9c66-73be22c8da5c',
    'ECO_POWER': '1ed14e6f-02ef-4696-ba51-e3d6fc457596',
    'BATT_ENG_ERROR': '18fc9300-34b0-4877-897d-3a3f58dbfc96',
    'REGEN': '1d9bedd3-5fe2-47ef-a052-bf86641cb965',
    'CRUISE_THROTTLE': 'fbdaa58c-fa30-4fda-adac-d2cf7f3a3593',
    'CRUISE_SET_SPEED': '7c5ffcfa-4b0a-4904-966a-0fd9476ae3fd',
    'RPM': 'a50c1d59-8736-428d-8268-fff766468aaa',
    'OVERTEMP_E_RM': '5251493b-dad7-4c23-94c0-3392b8b75ac9',
    'OVERTEMP_E_LM': '8ea6acfa-03a1-4945-888d-02a6b9d2a861',
    'OVERTEMP_E_RC': '264d925b-b872-41e9-b6ec-215623ba706b',
    'OVERTEMP_E_LC': '347c4f98-f731-438c-bed7-f8ceeb1fc7a3',
    'OVERTEMP_W_RM': 'aefd0582-a165-4490-8780-d7e9a950b9ac',
    'OVERTEMP_W_LM': '810f92e9-9238-4f94-9c54-ebe4b978690a',
    'OVERTEMP_W_RC': '7d2222dd-0ca2-499f-b977-21fe0d037221',
    'OVERTEMP_W_LC': '0daf20b0-b1de-4a99-96d8-30cd66a8af9c',
    'NGL5_CP_VENT_REQUEST': 'f482cfe8-c889-4f13-8c64-046315e14812',
    'NGL5_CLEAR_ERROR_LATCH': '0e3a3219-4f0b-43bd-b68d-a601cb5fc4b8',
    'NGL5_CAN_ENABLE': '69b8363f-ddeb-48bb-a01b-76b599b53828',
    'NGL5_MAX_I_AC': 'b87f0148-01cc-4f23-926b-701506bfb07c',
    'NGL5_MAX_V_DC': 'b7e974ff-007a-4eee-880a-87893bb9752c',
    'NGL5_MAX_I_DC': 'f385fce7-ecfe-4c72-bb42-2995e23b5ee6',
    'TIME': '28a793c8-6aa6-4578-a807-529beb121230',
    'OVARTECH_MEAS_V_DC': 'd47bdccf-b982-4fae-be28-d950323f692a',
    'OVARTECH_MEAS_I_DC': '7bf13a8b-8d98-4263-8f89-629d8775e280',
    'OVARTECH_COMM_TIMEOUT': 'f1b739e5-a22b-4ca8-9511-7bd503606288',
    'OVARTECH_CHARGER_OFF': 'f65b074e-a84d-4ca6-8cff-f5a115e1b370',
    'OVARTECH_AC_ERROR': 'c5405275-c337-4953-a01a-2f1c4970d8ca',
    'OVARTECH_OVERTEMP': '63b39ab4-29bf-4fdb-bdc8-7e837317e14b',
    'OVARTECH_HW_FAIL': '7bec3451-0a77-4a57-aa6a-676b3953763e',
    'NGL5_MEAS_I_AC': '8f204833-3ade-4f25-b5b0-a0878a40543a',
    'NGL5_MEAS_V_AC': '3ac6c38e-d6ec-4e98-9b1f-611244842349',
    'NGL5_MEAS_V_DC': '178820c1-f1c7-4256-b56c-078b4fa0ad3a',
    'NGL5_MEAS_I_DC': '76eab3ae-e580-4fc4-8a21-71445ce13492',
    'NGL5_MAINS_US_L2': 'c3464990-aed2-440d-8d1b-63573cd5ce09',
    'NGL5_MAINS_US_L1': '4b1a56ab-7514-40a0-8cf6-713d32b170bb',
    'NGL5_MAINS_EUROPE': 'fb4e07cc-8a3e-49de-bed1-d277c9cd60b3',
    'NGL5_FAN_ACTIVE': '33862715-5d5f-49ce-ad9c-12372a977a2e',
    'NGL5_GENERAL_WARNING': 'ebca14eb-be86-458b-8277-034d5a22f90f',
    'NGL5_GENERAL_ERROR': 'cebf4f12-521d-477f-a5bf-abb072e1c160',
    'NGL5_POWER_ENABLE': 'b95c270e-1122-49b8-8ec2-feae7aaa812b',
    'TEMP_RIGHT_MOTOR': '968a3278-d0b7-4ed6-af2f-a6d7440371f4',
    'TEMP_LEFT_MOTOR': 'bcf7e6a8-587f-42ae-a014-049a6c003b36',
    'TEMP_RIGHT_DRIVER': '6826de2c-89f6-4bd1-aeea-042e5d7bfde7',
    'TEMP_LEFT_DRIVER': '4269a7b5-8e30-4b05-a7d6-905d264b9bc5',
    'PRESSURE_FRONT_LEFT': '1acd8367-7ac2-46c9-a778-7a3d2084f2b6',
    'PRESSURE_FRONT_RIGHT': 'c5922471-27c1-458f-8129-9b1766f80862',
    'PRESSURE_REAR_LEFT': 'e0f8b8c5-c631-4a42-8264-b4be9706783e',
    'PRESSURE_REAR_RIGHT': '7713326e-4785-46e8-a9eb-9fabd97300f8'
}

